jupyter_sphinx/__init__.py

```diff
@@ -170,15 +179,28 @@ def setup(app):
         man=(skip, None),
     )
 
-    # JupyterCellNode is a container that holds the input and
-    # any output, so we render it as a container.
+    # Register our container nodes, these should behave just like a regular container
+    for node in [JupyterCellNode, CellInputNode, CellOutputNode]:
+        app.add_node(
+            node,
+            override=True,
+            html=(render_container),
+            latex=(render_container),
+            textinfo=(render_container),
+            text=(render_container),
+            man=(render_container),
+        )
+
+    # Register the output bundle node.
+    # No translators should touch this node because we'll replace it in a post-transform
     app.add_node(
-        JupyterCellNode,
-        html=render_container,
-        latex=render_container,
-        textinfo=render_container,
-        text=render_container,
-        man=render_container,
+        CellOutputBundleNode,
+        override=True,
+        html=(skip, None),
```

Comment:
Would it make sense to add a `raise` transform that would raise a sphinx extension error, for catching unexpected cases when the transform didn't run?

Reply:
I'm not sure what a raise translator is, but my intuition is that this would halt the build if a builder that wasn't supported was used? I feel like that'd be extreme no?

Reply:
This would be to clearly signal, that the node isn't meant to appear in the finalized AST. Seems to make sense and might provide an early notification when something goes awry.

Reply:
Ahh good point, OK I get it. Would a `raise` translator be something built-in to sphinx? or something we define?

Reply:
I think it'd be similar to `skip`, which we define in the same file:

```python
def halt(self, node):
    raise sphinx.errors.ExtensionError('Unexpected node encountered ...')
```

I don't expect that sphinx itself would define this because we should be raising an extension error, I think.

Reply:
ok, I think the latest push adds this, let me know if that's not what you had in mind.

What are your thoughts about testing?

Reply:
Thanks, this looks good.

I think raising an error when that node is attached is largely an internal consistency check, more akin to an assertion. Therefore I think it is appropriate that it is not covered by tests.